### PR TITLE
CLEWS-16687 Use contains endpoint for get_descendant_regions

### PR DESCRIPTION
**Pull request by itsjohnward**
Switched over get_descendant_regions to use the new `/v2/regions/contains` functionality added in v1.60 instead of recursively checking the 'contains' of each region.

Since `/v2/regions/contains` only returns ids where the old method returned entire objects, I had to add in `lookup()`s to make the output identical. So, by default, there's not an order of magnitude performance difference between the old way and the new way.

But, I added this `include_details` flag which bypasses that and does have an order of magnitude speed-up.

To test the speed difference I ran the following script:

```py
from api.client.gro_client import GroClient
import os
import time

client = GroClient('api.gro-intelligence.com', os.environ['GROAPI_TOKEN'])
start = time.time()
client.get_descendant_regions(1215, descendant_level=5)
end = time.time()
print('{:.2f}s'.format(end-start))
```

1. Before this change: 327.72s
2. With this change: 257.62s
3. With this change, with `include_details=False` added to the params list: 0.15s

Case 2 can be sped up in the future by batching those lookup() requests, but case 3 is immediately usable and much improved.

### CODE REVIEW

**Comment by JGrinst** [comment]
What's the benefit to having `include_details=True` as the default option. We assume that users will want id + name?

**Comment by itsjohnward** [comment]
> What's the benefit to having `include_details=True` as the default option. We assume that users will want id + name?

Made it the default in order to match the old output and not break anyone relying on it. For backwards compatibility's sake.

For example, if someone had previously written the code

```py
positions = [(region['latitude'], region['longitude'])
             for region in client.get_descendant_regions(1215, descendant_level=5)]
```

in order to get the lat/lons of all the counties in the US, that would work. But if I made `include_details=False` the default then when they updated their client their previously-functional code would start throwing

```py
Traceback (most recent call last):
    KeyError: 'latitude'
```

File: api/client/lib.py [hunk 1]
```diff
@@ -599,19 +599,31 @@ def get_geojson(access_token, api_host, region_id):
...
+        return [{'id': descendant_region_id}
+                for descendant_region_id in descendant_region_ids]
+
+    detailed_regions = [lookup(access_token, api_host, 'regions', descendant_region_id)
+                        for descendant_region_id in descendant_region_ids]
+    if not include_historical:
+        # filter out regions with the 'historical' flag
+        return [detailed_region for detailed_region in detailed_regions
+                if not detailed_region['historical']]
+    return detailed_regions
```
Comment (seungatGro) [inline api/client/lib.py]: minor. running with include_historical=False, include_details=False includes details in response

**Review by seungatGro — CHANGES_REQUESTED** [review]
minor comment on include_details=False response. otherwise LGTM

File: api/client/lib.py (hunk 1 above)
Comment (itsjohnward) [reply api/client/lib.py]: Updated.

```py
>>> client.get_descendant_regions(12, descendant_level=3, include_historical=True, include_details=False)
[{'id': 1001}, {'id': 1011}, {'id': 1015}, {'id': 1017}, {'id': 1018}, {'id': 1025}, {'id': 1031}, {'id': 1035}, {'id': 1053}, {'id': 1075}, {'id': 1094}, {'id': 1095}, {'id': 1096}, {'id': 1097}, {'id': 1100}, {'id': 1103}, {'id': 1105}, {'id': 1106}, {'id': 1109}, {'id': 1110}, {'id': 1111}, {'id': 1113}, {'id': 1120}, {'id': 1123}, {'id': 1124}, {'id': 1136}, {'id': 1140}, {'id': 1143}, {'id': 1154}, {'id': 1155}, {'id': 1161}, {'id': 1165}, {'id': 1166}, {'id': 1178}, {'id': 1182}, {'id': 1189}, {'id': 1195}, {'id': 1196}, {'id': 1197}, {'id': 1199}, {'id': 1205}, {'id': 1206}, {'id': 1211}, {'id': 1217}, {'id': 1220}, {'id': 1223}, {'id': 1229}, {'id': 1231}, {'id': 1232}, {'id': 1240}, {'id': 1247}, {'id': 1250}, {'id': 1254}, {'id': 1257}, {'id': 1270}, {'id': 1271}, {'id': 1272}, {'id': 1273}, {'id': 1277}]
```

```py
>>> client.get_descendant_regions(12, descendant_level=3, include_historical=False, include_details=False)
[{'id': 1001}, {'id': 1011}, {'id': 1015}, {'id': 1017}, {'id': 1018}, {'id': 1025}, {'id': 1031}, {'id': 1035}, {'id': 1053}, {'id': 1075}, {'id': 1094}, {'id': 1095}, {'id': 1096}, {'id': 1097}, {'id': 1100}, {'id': 1103}, {'id': 1105}, {'id': 1106}, {'id': 1109}, {'id': 1110}, {'id': 1111}, {'id': 1113}, {'id': 1120}, {'id': 1123}, {'id': 1124}, {'id': 1136}, {'id': 1140}, {'id': 1143}, {'id': 1154}, {'id': 1155}, {'id': 1161}, {'id': 1165}, {'id': 1166}, {'id': 1178}, {'id': 1182}, {'id': 1189}, {'id': 1195}, {'id': 1196}, {'id': 1197}, {'id': 1199}, {'id': 1205}, {'id': 1206}, {'id': 1211}, {'id': 1217}, {'id': 1220}, {'id': 1223}, {'id': 1229}, {'id': 1231}, {'id': 1232}, {'id': 1240}, {'id': 1247}, {'id': 1250}, {'id': 1254}, {'id': 1257}, {'id': 1277}]
```

```py
>>> client.get_descendant_regions(12, descendant_level=3, include_historical=False, include_details=True)
[{'contains': [10022, 10004, 10012, 10021, 10025, 13426, 10030, 10010, 10024, 13425, 10015, 10023, 10020, 10026, 10006, 10032, 10016, 10014, 10011, 10013, 10005, 10001, 10027, 10028, 10029, 10019, 10031, 10007, 10003, 10018, 10009, 10017, 10008, 10002], 'definition': None, 'historical': False, 'id': 1001, 'latitude': 33.8284, 'level': 3, 'longitude': 66.0296, 'name': 'Afghanistan'}, {'contains': [10168, 10170, 10163, 10166, 10161, 10160, 10167, 10164, 10165, 10169, 10162], 'definition': None, 'historical': False, 'id': 1011, 'latitude': 40.2932, 'level': 3, 'longitude': 44.9381, 'name': 'Armenia'}, {'contains': [10229, 10244, 10236, 10206, 10203, 10231, 10241, 10220, 10253, 10202, 10214, 10228, 10257, 10200, 10227, 10218, 10215, 10205, 10245, 10238, 10201, 10221, 10193, 10192, 10247, 10255, 10194, 10246, 10254, 10230, 10239, 10250, 10199, 10259, 10232, 10212, 10223, 10219, 10251, 10242, 10252, 10233, 10213, 10225, 10198, 10196, 10234, 10209, 10210, 10204, 10240, 10216, 10224, 10207, 10197, 10248, 10256, 10208, 10191, 10217, 10258, 10243, 10195, 10260, 10226, 10237, 10249, 10235, 10211, 10222], 'definition': None, 'historical': False, 'id': 1015, 'latitude': 40.3044, 'level': 3, 'longitude': 47.5136, 'name': 'Azerbaijan'}, {'contains': [10296, 10295, 10293, 10294, 10297], 'definition': None, 'historical': False, 'id': 1017, 'latitude': 26.0663, 'level': 3, 'longitude': 50.5427, 'name': 'Bahrain'}, ...
```

**Review by seungatGro — APPROVED** [review]
changes LGTM

**Comment by itsjohnward** [comment]
FYI to reviewers, I had to start locking packages in the test-requirements file to versions that still support Python 2. `statsmodels` released their v0.11.0 today https://github.com/statsmodels/statsmodels/releases/tag/v0.11.0 which dropped python 2 support. Latest version is brought in by `tsfresh`, so shippable was failing on python 2.7 until https://github.com/gro-intelligence/api-client/pull/157/commits/7443101ec3dc444b54fa0afbf097d00d41f4ec44

**Pull request merged**